### PR TITLE
feat(parser): add support for parsing SQS events wrapped in Kinesis Firehose

### LINKED ISSUE #2293: Feature request: Parsing SQS events wrapped in Kinesis Firehose events
### Discussed in https://github.com/awslabs/aws-lambda-powertools-python/discussions/2287

<div type='discussions-op-text'>

<sup>Originally posted by **sk0g** May 18, 2023</sup>
So we have payloads that are essentially Kinesis Firehouse events containing an SQS event. Is there a way to unwrap this using `@event_parser()`? If not, what's the next best thing?</div>

### PR DESCRIPTION
**Pull request by leandrodamascena**
**Issue number:** #2293 

### Changes

In this update, we enhance the parser to handle SQS events that are wrapped in Kinesis Firehose. 

parser_sqs

### User experience

With this new Model, you can easily extract relevant information from SQS events within the Kinesis Firehose ecosystem.

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [x] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

**Comment by rubenfonseca** [comment]
Looking at this now

File: aws_lambda_powertools/utilities/parser/models/kinesis_firehose_sqs.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+class KinesisFirehoseSqsRecord(BaseModel):
+    data: SqsRecordModel
+    recordId: str
+    approximateArrivalTimestamp: PositiveInt
+    kinesisRecordMetadata: Optional[KinesisFirehoseRecordMetadata]
+
+    @validator("data", pre=True, allow_reuse=True)
+    def data_base64_decode(cls, value):
+        # Firehose payload is encoded twice
```
Comment (rubenfonseca) [inline aws_lambda_powertools/utilities/parser/models/kinesis_firehose_sqs.py]: 😱

**Review by rubenfonseca — APPROVED** [review]
Awesome job, the double encoding is really scary :D

File: aws_lambda_powertools/utilities/parser/models/kinesis_firehose_sqs.py (hunk 1 above)
Comment (leandrodamascena) [reply aws_lambda_powertools/utilities/parser/models/kinesis_firehose_sqs.py]: Ruben, I will validate this one more time before merging. But scare me too.

**Comment by leandrodamascena** [comment]
> Awesome job, the double encoding is really scary :D

Ruben, your comment confused me about this and I investigated again. I was using the wrong event coming from EventBridge and indeed it's encoded once and not twice. Sorry about that, now it's ok!

**Review by rubenfonseca — APPROVED** [review]
LGTM!

**Pull request merged**